# DEL ACERO AL ALGORITMO
## Estadística Descriptiva aplicada a la Educación Superior del Biobío

**Base de datos:** Matrículas de Educación Superior de la Región del Biobío, 2021.

### Pregunta central
**¿Desde qué base de capital humano parte el Biobío para transformar su histórica vocación industrial hacia una economía de manufactura avanzada e Industria 4.0?**

### ¿Qué significa el título “Del acero al algoritmo”?

El título funciona como una **metáfora del problema que queremos estudiar**.

- **“Acero”** representa la base productiva e industrial del Biobío y, dentro de este trabajo, se relaciona con los **motores productivos**: industria y manufactura, construcción e infraestructura, logística y puertos, forestal y madera, pesca y acuicultura, y agroalimentación.
- **“Algoritmo”** representa las **capacidades transformadoras** asociadas a una economía más tecnológica: Digital y TIC, automatización y robótica, y energía, sustentabilidad y biotecnología.

La idea **no es afirmar que el Biobío esté abandonando su industria ni demostrar que ya exista una transición causada por la tecnología**. Con esta base solo observamos una fotografía de la matrícula de educación superior de 2021. Por eso, el título plantea una pregunta de análisis: **qué peso tienen la formación vinculada a la base productiva y las capacidades tecnológicas que podrían complementarla**.

En simple, el proyecto busca responder: **¿con qué formación de capital humano contaba el Biobío en 2021 para combinar su identidad productiva con nuevas capacidades tecnológicas?**

El notebook se desarrolla con las herramientas trabajadas en clases y se concentra en los análisis que aportan directamente al trabajo.


# 0. Carga y preparación de la base

Primero cargamos la base, revisamos su estructura y preparamos las tres unidades que vamos a utilizar: registros originales, matrículas de pregrado y ofertas académicas únicas.


In [1]:
#Importamos Pandas y cargamos el archivo Excel
import pandas as pd

df = pd.read_excel(
    '11_MATRICULAS_ED_SUPERIOR_BIOBIO_2021.xlsx',
    sheet_name='BASE DE DATOS'
)

#Mostramos las primeras filas para comprobar la carga
df.head()


,ID,GENERO,EDAD,RANGO EDAD,AÑO INGRESO,SEMESTRE INGRESO,TIPO DE INSTITUCION,NOMBRE DE INSTITUCION,ACREDITACION INSTITUCIONAL,PERIODO DE ACREDITACION,...,NIVEL CARRERA,AREA CONOCIMIENTO,DURACION PLAN DE ESTUDIO (SEMESTRES),DURACION PROCESO TITULACION (SEMESTRES),DURACION TOTAL CARRERA (SEMESTRES),VALOR MATRICULA (PESOS),VALOR ARANCEL (PESOS),REGION SEDE,PROVINCIA SEDE,COMUNA SEDE
0,52,Femenino,24,20 a 24,2016,Primer semestre,Universidades CRUCH,UNIVERSIDAD DE CONCEPCION,ACREDITADA,11/11/2016 AL 11/11/2023,...,Carreras Profesionales,Salud,10,0,10,120000,4243207,Biobio,Concepcion,Concepcion
1,58,Masculino,49,40 y mas,2019,Primer semestre,Institutos Profesionales,IP AIEP,ACREDITADA,22/10/2017 AL 22/10/2022,...,Carreras Tecnicas,Tecnologia,5,1,5,190000,1840000,Biobio,Concepcion,Concepcion
2,73,Masculino,21,20 a 24,2021,Primer semestre,Universidades Privadas,UNIVERSIDAD DEL DESARROLLO,ACREDITADA,15/12/2016 AL 15/12/2021,...,Carreras Profesionales,Derecho,10,2,12,494445,5351640,Biobio,Concepcion,Concepcion
3,80,Femenino,24,20 a 24,2020,Primer semestre,Centros de Formacion Tecnica,CFT LOTA-ARAUCO,ACREDITADA,21/07/2019 AL 21/07/2023,...,Carreras Tecnicas,Tecnologia,5,1,5,30000,1450000,Biobio,Concepcion,Lota
4,88,Femenino,25,25 a 29,2018,Primer semestre,Universidades CRUCH,UNIVERSIDAD DEL BIO-BIO,ACREDITADA,28/08/2019 AL 28/08/2024,...,Carreras Profesionales,Administracion y Comercio,10,0,10,105000,3255000,Biobio,Concepcion,Concepcion


**Explicación:** En esta celda importamos la biblioteca pandas, que es la que usamos para trabajar con tablas de datos en Python. Después leemos el archivo Excel y guardamos la hoja **BASE DE DATOS** dentro del DataFrame df. Finalmente usamos head() para revisar las primeras cinco filas y comprobar que la información se cargó correctamente. En esas filas ya se pueden ver variables que después serán importantes, como género, edad, tipo de institución, carrera, área del conocimiento, arancel y provincia de sede.


In [2]:
#Revisamos la cantidad de filas y columnas
df.shape


(106555, 28)

**Explicación:** Aquí usamos shape para conocer el tamaño de la base original. El resultado **(106555, 28)** significa que tenemos **106.555 registros** y **28 columnas**. Este dato es importante porque define el tamaño inicial de la información antes de aplicar cualquier filtro o crear subbases para el análisis.


In [3]:
#Revisamos las columnas, tipos de datos y valores no nulos
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 106555 entries, 0 to 106554
Data columns (total 28 columns):
 #   Column                                   Non-Null Count   Dtype  
---  ------                                   --------------   -----  
 0   ID                                       106555 non-null  int64  
 1   GENERO                                   106555 non-null  str    
 2   EDAD                                     106555 non-null  int64  
 3   RANGO EDAD                               106555 non-null  str    
 4   AÑO INGRESO                              106555 non-null  int64  
 5   SEMESTRE INGRESO                         106555 non-null  str    
 6   TIPO DE INSTITUCION                      106555 non-null  str    
 7   NOMBRE DE INSTITUCION                    106555 non-null  str    
 8   ACREDITACION INSTITUCIONAL               106555 non-null  str    
 9   PERIODO DE ACREDITACION                  102840 non-null  str    
 10  AÑOS DE ACREDITACION                     10

**Explicación:** Con info() revisamos la estructura completa del DataFrame. La salida confirma que existen 106.555 filas y 28 columnas, además muestra el tipo de dato de cada variable. También permite detectar valores faltantes. En este caso, las columnas **PERIODO DE ACREDITACION** y **AÑOS DE ACREDITACION** tienen menos registros completos que el resto. Las variables principales que usamos en este trabajo, como edad, carrera, área, arancel, institución y provincia, sí tienen información completa en la base.


In [4]:
#Controlamos duplicados completos y valores nulos
duplicados_completos = int(df.duplicated().sum())
nulos_por_columna = df.isna().sum()
nulos_relevantes = nulos_por_columna[nulos_por_columna > 0]

print('Duplicados completos:', duplicados_completos)
print('\nValores nulos por columna:')
print(nulos_relevantes)


Duplicados completos: 0

Valores nulos por columna:
PERIODO DE ACREDITACION    3715
AÑOS DE ACREDITACION       3715
dtype: int64


**Control de calidad:** comprobamos de forma explícita los duplicados completos y los valores nulos. La base tiene **0 filas completamente duplicadas**. Los valores faltantes se concentran en variables de acreditación, que no son necesarias para responder las tres preguntas del trabajo. Más adelante, para el análisis de precios, también excluimos los **26 registros con arancel igual a $0**.


In [5]:
#Contamos los niveles de carrera presentes en la base
df['NIVEL CARRERA'].value_counts()


NIVEL CARRERA
Carreras Profesionales    75855
Carreras Tecnicas         25238
Magister                   3340
Postitulo                  1710
Doctorado                   412
Name: count, dtype: int64

**Explicación:** En esta celda contamos cuántos registros existen para cada nivel de carrera usando value_counts(). La mayor cantidad corresponde a **Carreras Profesionales (75.855)** y **Carreras Técnicas (25.238)**. También aparecen Magíster, Postítulo y Doctorado, pero estos niveles no forman parte del análisis principal porque el trabajo se concentra en la formación de pregrado. Por eso, en el siguiente paso utilizamos solamente carreras profesionales y técnicas.


La base original contiene **106.555 registros y 28 columnas**. Para el análisis principal nos interesa trabajar con carreras de pregrado.


In [6]:
#Definimos los niveles de pregrado
pregrado = ['Carreras Profesionales', 'Carreras Tecnicas']

#Filtramos las matrículas de pregrado
d = df[
    df['NIVEL CARRERA'].isin(pregrado)
].copy()

#Para analizar precios dejamos solo aranceles mayores que $0
d_precio = d[
    d['VALOR ARANCEL (PESOS)'] > 0
].copy()

print('Matrículas de pregrado:', d.shape[0])
print('Matrículas con arancel mayor que $0:', d_precio.shape[0])


Matrículas de pregrado: 101093
Matrículas con arancel mayor que $0: 101067


**Explicación:** Aquí preparamos las dos subbases principales del trabajo. Primero definimos como pregrado las **Carreras Profesionales** y **Carreras Técnicas**, y filtramos esos registros en el DataFrame d. Con ese filtro quedan **101.093 matrículas de pregrado**. Luego creamos d_precio, donde además exigimos que el arancel sea mayor que $0. Esta segunda base queda con **101.067 registros** y será utilizada cuando analicemos precios. La diferencia es de 26 registros con arancel igual a cero, que no usamos como precio porque podrían distorsionar los cálculos.


## Oferta académica única

Cuando analizamos precios no conviene repetir el mismo arancel una vez por cada estudiante matriculado. Por eso creamos una base donde cada oferta académica aparece una sola vez.


In [7]:
#Columnas que identifican una oferta académica
clave_oferta = [
    'NOMBRE DE INSTITUCION',
    'NOMBRE CARRERA',
    'COMUNA SEDE',
    'MODALIDAD',
    'JORNADA',
    'DURACION TOTAL CARRERA (SEMESTRES)',
    'VALOR MATRICULA (PESOS)',
    'VALOR ARANCEL (PESOS)'
]

#Eliminamos las ofertas repetidas
ofertas = d_precio.drop_duplicates(
    subset=clave_oferta
).copy()

print('Ofertas académicas únicas:', ofertas.shape[0])


Ofertas académicas únicas: 1273


**Explicación:** Esta celda crea la base de **ofertas académicas únicas**. La base original está a nivel de matrícula, por lo que una misma carrera puede aparecer muchas veces porque tiene muchos estudiantes. Si analizáramos precios directamente desde esas filas, un mismo arancel se repetiría una vez por cada estudiante. Para evitar eso usamos drop_duplicates() considerando institución, carrera, comuna, modalidad, jornada, duración, matrícula y arancel. El resultado es **1.273 ofertas únicas**, que será la unidad que utilizaremos para comparar aranceles entre carreras y áreas.


# 1. Población, muestra y variables

**Población:** matrículas de educación superior de la Región del Biobío durante 2021.

**Base disponible:** 106.555 registros.

**Subbase principal:** 101.093 matrículas de pregrado.

## 1.1 Clasificación de las variables principales

Siguiendo el criterio trabajado en el **Laboratorio 1**, distinguimos variables cualitativas y cuantitativas. Dentro de ellas usamos los subtipos nominal, ordinal, discreta y continua.

| Variable | Tipo general | Subtipo | Justificación breve |
|---|---|---|---|
| GÉNERO | Cualitativa | Nominal | Son categorías sin un orden natural. |
| RANGO EDAD | Cualitativa | Ordinal | Los rangos tienen un orden natural de menor a mayor edad. |
| TIPO DE INSTITUCIÓN | Cualitativa | Nominal | Distingue categorías de institución sin jerarquía numérica. |
| NOMBRE CARRERA | Cualitativa | Nominal | Identifica categorías de carreras sin orden natural. |
| ÁREA DEL CONOCIMIENTO | Cualitativa | Nominal | Clasifica áreas sin un orden natural. |
| PROVINCIA SEDE | Cualitativa | Nominal | Identifica territorios sin un orden estadístico. |
| EDAD | Cuantitativa | Continua | Seguimos el criterio del Laboratorio 1 del profesor, donde edad se clasifica como continua. |
| AÑO INGRESO | Cuantitativa | Discreta | Se registra en años enteros, por ejemplo 2019, 2020 o 2021. |
| DURACIÓN TOTAL CARRERA (SEMESTRES) | Cuantitativa | Continua | Representa una medida de duración expresada en semestres. |
| VALOR ARANCEL (PESOS) | Cuantitativa | Continua | Es una magnitud numérica sobre la que calculamos media, mediana, percentiles y dispersión. |

**Cómo recordarlo:** nominal = categorías sin orden; ordinal = categorías ordenadas; discreta = valores contables separados; continua = una medición sobre una escala numérica.


# 2. Descripción general de la base

Antes de responder las preguntas del trabajo revisamos algunas características generales usando tablas de frecuencia, gráficos y medidas descriptivas.


## 2.1 Género


In [8]:
#Calculamos la frecuencia absoluta
cuenta_genero = d.groupby('GENERO').size()

#Creamos la tabla con frecuencia absoluta y relativa
tabla_genero = pd.DataFrame({
    'Frecuencia absoluta': cuenta_genero,
    'Frecuencia relativa (%)': round(cuenta_genero/d.shape[0] * 100, 1)
})

tabla_genero


,Frecuencia absoluta,Frecuencia relativa (%)
GENERO,,
Femenino,54881,54.3
Masculino,46212,45.7


**Explicación:** Aquí construimos una tabla de frecuencias para la variable género. Primero contamos cuántas matrículas hay en cada categoría y luego calculamos qué porcentaje representa cada una respecto de las 101.093 matrículas de pregrado. El resultado muestra **54.881 matrículas femeninas (54,3%)** y **46.212 masculinas (45,7%)**. Esto indica que en la base existe una participación femenina algo mayor, aunque ambos grupos tienen una presencia importante.


En la base de pregrado, las matrículas femeninas representan **54,3%** y las masculinas **45,7%**.


## 2.2 Área del conocimiento


In [9]:
#Calculamos la frecuencia por área del conocimiento
cuenta_area = d.groupby('AREA CONOCIMIENTO').size()

#Creamos la tabla y la ordenamos de mayor a menor
tabla_area = pd.DataFrame({
    'Frecuencia absoluta': cuenta_area,
    'Frecuencia relativa (%)': round(cuenta_area/d.shape[0] * 100, 1)
})

tabla_area = tabla_area.sort_values(
    'Frecuencia absoluta',
    ascending=False
)

tabla_area


,Frecuencia absoluta,Frecuencia relativa (%)
AREA CONOCIMIENTO,,
Tecnologia,27866,27.6
Salud,24597,24.3
Administracion y Comercio,13526,13.4
Educacion,11480,11.4
Ciencias Sociales,10677,10.6
Arte y Arquitectura,3611,3.6
Derecho,3529,3.5
Ciencias Basicas,2936,2.9
Agropecuaria,2361,2.3


**Explicación:** En esta celda agrupamos las matrículas por **área del conocimiento** y calculamos frecuencia absoluta y relativa. Después ordenamos de mayor a menor para identificar rápidamente las áreas con mayor presencia. **Tecnología** ocupa el primer lugar con **27.866 matrículas (27,6%)**, seguida por **Salud** con **24.597 (24,3%)**. Administración y Comercio aparece con 13,4% y Educación con 11,4%. Esta tabla nos entrega una primera fotografía de qué áreas concentran más estudiantes en el Biobío.


Las áreas con mayor cantidad de matrículas son **Tecnología** y **Salud**.


## 2.3 Edad agrupada


In [10]:
#Agrupamos la edad en intervalos
d['Intervalos_edad'] = pd.cut(
    d['EDAD'],
    bins=10,
    include_lowest=True,
    precision=0
)

#Calculamos las frecuencias
cuenta_edad = d.groupby(
    'Intervalos_edad',
    observed=True
).size()

tabla_edad = pd.DataFrame({
    'Frecuencia absoluta': cuenta_edad,
    'Frecuencia relativa (%)': cuenta_edad/d.shape[0] * 100
})

#Como los intervalos están ordenados, agregamos las frecuencias acumuladas
tabla_edad['Frecuencia acumulada'] = tabla_edad['Frecuencia absoluta'].cumsum()
tabla_edad['Frecuencia relativa acumulada (%)'] = tabla_edad['Frecuencia relativa (%)'].cumsum()

round(tabla_edad, 1)


,Frecuencia absoluta,Frecuencia relativa (%),Frecuencia acumulada,Frecuencia relativa acumulada (%)
Intervalos_edad,,,,
"(16.0, 23.0]",48400,47.9,48400,47.9
"(23.0, 29.0]",35920,35.5,84320,83.4
"(29.0, 35.0]",9489,9.4,93809,92.8
"(35.0, 41.0]",3706,3.7,97515,96.5
"(41.0, 46.0]",2061,2.0,99576,98.5
"(46.0, 52.0]",1072,1.1,100648,99.6
"(52.0, 58.0]",362,0.4,101010,99.9
"(58.0, 64.0]",73,0.1,101083,100.0
"(64.0, 70.0]",7,0.0,101090,100.0


**Explicación:** Como la edad es una variable cuantitativa con muchos valores posibles, la agrupamos en **10 intervalos** mediante pd.cut(). Luego calculamos frecuencia absoluta, frecuencia relativa y, porque los intervalos tienen un orden natural, agregamos la **frecuencia acumulada** con cumsum(). El grupo aproximado de **17 a 23 años concentra 47,9%** de las matrículas y el siguiente, de **23 a 29 años, 35,5%**. La acumulada permite ver cuánto del total se reúne al avanzar por los intervalos de edad.


## 2.4 Arancel agrupado


In [11]:
#Agrupamos los aranceles en intervalos
d_precio['Intervalos_arancel'] = pd.cut(
    d_precio['VALOR ARANCEL (PESOS)'],
    bins=9,
    include_lowest=True,
    precision=0
)

#Calculamos las frecuencias
cuenta_arancel = d_precio.groupby(
    'Intervalos_arancel',
    observed=True
).size()

tabla_arancel = pd.DataFrame({
    'Frecuencia absoluta': cuenta_arancel,
    'Frecuencia relativa (%)': cuenta_arancel/d_precio.shape[0] * 100
})

#Como los intervalos están ordenados, agregamos las frecuencias acumuladas
tabla_arancel['Frecuencia acumulada'] = tabla_arancel['Frecuencia absoluta'].cumsum()
tabla_arancel['Frecuencia relativa acumulada (%)'] = tabla_arancel['Frecuencia relativa (%)'].cumsum()

round(tabla_arancel, 1)


,Frecuencia absoluta,Frecuencia relativa (%),Frecuencia acumulada,Frecuencia relativa acumulada (%)
Intervalos_arancel,,,,
"(641865.0, 1553741.0]",10846,10.7,10846,10.7
"(1553741.0, 2457482.0]",32641,32.3,43487,43.0
"(2457482.0, 3361223.0]",11719,11.6,55206,54.6
"(3361223.0, 4264964.0]",20615,20.4,75821,75.0
"(4264964.0, 5168706.0]",17271,17.1,93092,92.1
"(5168706.0, 6072447.0]",4193,4.1,97285,96.3
"(6072447.0, 6976188.0]",1462,1.4,98747,97.7
"(6976188.0, 7879929.0]",1257,1.2,100004,98.9
"(7879929.0, 8783670.0]",1063,1.1,101067,100.0


**Explicación:** Aplicamos el mismo procedimiento al arancel, agrupándolo en nueve intervalos. Calculamos frecuencia absoluta, relativa y, al tratarse de intervalos ordenados, **frecuencia acumulada** mediante cumsum(). El intervalo con mayor frecuencia está aproximadamente entre **$1,55 y $2,46 millones**, con **32,3%** de los registros. También existe una concentración importante entre aproximadamente **$3,36 y $4,26 millones (20,4%)** y entre **$4,26 y $5,17 millones (17,1%)**.


## 2.5 Gráficos principales


In [12]:
#Importamos la biblioteca para crear gráficos
import matplotlib.pyplot as plt


**Explicación:** Aquí importamos matplotlib.pyplot, que es la biblioteca que usamos para construir los gráficos. La abreviamos como plt para utilizarla de manera más simple en las siguientes celdas. Esta celda no genera un resultado visible porque solamente deja preparada la herramienta de visualización.


In [13]:
#Gráfico de barras por área del conocimiento
fig, ax = plt.subplots(figsize=(10,5))

ax.bar(
    tabla_area.index,
    tabla_area['Frecuencia absoluta'],
    edgecolor='black'
)

ax.set_title('Matrículas por área del conocimiento')
ax.set_xlabel('Área del conocimiento')
ax.set_ylabel('Cantidad de matrículas')
plt.xticks(rotation=45, ha='right')

plt.show()


**Explicación:** Este gráfico representa la cantidad de matrículas por área del conocimiento. Usamos barras porque permiten comparar fácilmente categorías distintas. Al observar las alturas se distingue con claridad que **Tecnología** y **Salud** son las áreas con mayor matrícula, mientras que Humanidades, Agropecuaria y Ciencias Básicas presentan cantidades bastante menores. El gráfico confirma visualmente el orden que ya habíamos visto en la tabla de frecuencias.


In [14]:
#Gráfico de la distribución de la edad
fig, ax = plt.subplots(figsize=(8,5))

ax.bar(
    tabla_edad.index.astype(str),
    tabla_edad['Frecuencia relativa (%)'],
    edgecolor='black',
    width=1
)

ax.set_title('Distribución de matrículas por edad')
ax.set_xlabel('Intervalos de edad')
ax.set_ylabel('Frecuencia relativa (%)')
plt.xticks(rotation=45, ha='right')

plt.show()


**Explicación:** Aquí graficamos los intervalos de edad utilizando la frecuencia relativa. La barra más alta corresponde al primer intervalo y la segunda al siguiente, lo que muestra que la matrícula se concentra principalmente en estudiantes jóvenes. A partir de los 30 años las barras disminuyen considerablemente. Este gráfico nos ayuda a ver de forma más rápida la forma de la distribución que antes habíamos observado en la tabla.


In [15]:
#Gráfico de la distribución del arancel
fig, ax = plt.subplots(figsize=(9,5))

ax.bar(
    tabla_arancel.index.astype(str),
    tabla_arancel['Frecuencia relativa (%)'],
    edgecolor='black',
    width=1
)

ax.set_title('Distribución de matrículas según arancel')
ax.set_xlabel('Intervalos de arancel')
ax.set_ylabel('Frecuencia relativa (%)')
plt.xticks(rotation=45, ha='right')

plt.show()


**Explicación:** Este gráfico muestra la distribución de las matrículas según los intervalos de arancel. Las barras permiten comparar qué rangos concentran más registros. Se observa que la mayor frecuencia aparece en los tramos medios, especialmente alrededor de $1,5 a $2,5 millones, aunque también existen grupos importantes en rangos superiores. Los intervalos más altos tienen menos registros, pero siguen presentes y después serán relevantes cuando estudiemos los aranceles sustantivamente caros.


Los gráficos muestran que la matrícula se concentra principalmente en **Tecnología y Salud**, en estudiantes de edades más jóvenes y en los rangos medios de arancel.


# 3. Medidas descriptivas

Usamos medidas de tendencia central y dispersión porque resumen las variables cuantitativas más importantes del trabajo.


## 3.1 Arancel


In [16]:
#Calculamos media, mediana y moda del arancel
media_arancel = d_precio['VALOR ARANCEL (PESOS)'].mean()
mediana_arancel = d_precio['VALOR ARANCEL (PESOS)'].median()
moda_arancel = d_precio['VALOR ARANCEL (PESOS)'].mode()[0]

print(f'Media: ${media_arancel:.0f}')
print(f'Mediana: ${mediana_arancel:.0f}')
print(f'Moda: ${moda_arancel:.0f}')


Media: $3224895
Mediana: $3133750
Moda: $3373000


**Explicación:** En esta celda calculamos tres medidas de tendencia central para el arancel. La **media** es aproximadamente **$3.224.895**, la **mediana $3.133.750** y la **moda $3.373.000**. La media representa el promedio, la mediana divide los registros en dos mitades y la moda corresponde al valor que más se repite. Como media y mediana son relativamente cercanas, ambas entregan una referencia útil del centro de la distribución, aunque sabemos que existen aranceles bastante más altos.


## 3.2 Edad


In [17]:
#Calculamos media, mediana y moda de la edad
media_edad = d['EDAD'].mean()
mediana_edad = d['EDAD'].median()
moda_edad = d['EDAD'].mode()[0]

print(f'Media: {media_edad:.1f} años')
print(f'Mediana: {mediana_edad:.0f} años')
print(f'Moda: {moda_edad} años')


Media: 24.4 años
Mediana: 23 años
Moda: 19 años


**Explicación:** Repetimos las medidas de tendencia central, pero ahora para la edad. La **media es 24,4 años**, la **mediana 23 años** y la **moda 19 años**. Esto significa que la mitad de las matrículas corresponde a estudiantes de 23 años o menos y que 19 años es la edad individual que más se repite. El resultado coincide con la tabla y el gráfico anteriores, donde vimos una fuerte concentración en edades jóvenes.


## 3.3 Dispersión del arancel


In [18]:
#Calculamos máximo y mínimo
maximo_arancel = d_precio['VALOR ARANCEL (PESOS)'].max()
minimo_arancel = d_precio['VALOR ARANCEL (PESOS)'].min()

#Calculamos el rango
rango_arancel = maximo_arancel - minimo_arancel

#Calculamos la desviación estándar
desviacion_arancel = d_precio['VALOR ARANCEL (PESOS)'].std()

#Calculamos el coeficiente de variación
cv_arancel = desviacion_arancel / media_arancel * 100

print(f'Rango: ${rango_arancel:.0f}')
print(f'Desviación estándar: ${desviacion_arancel:.0f}')
print(f'CV: {cv_arancel:.1f}%')

Rango: $8133670
Desviación estándar: $1518991
CV: 47.1%


**Explicación:** Aquí seguimos la forma del Laboratorio 5 del profesor para medir dispersión. Calculamos el **rango** entre el valor mínimo y máximo, la **desviación estándar** y el **coeficiente de variación**. El rango es de **$8.133.670**, la desviación estándar es aproximadamente **$1.518.991** y el coeficiente de variación es **47,1%**. Estas medidas muestran que los aranceles presentan una dispersión importante respecto de su promedio.

El arancel tiene una dispersión importante. El **coeficiente de variación es 47,1%**, por lo que los valores presentan bastante variación respecto del promedio.


# 4. Aplicación regional: Del acero al algoritmo

Para responder la pregunta central clasificamos las carreras en nueve familias. Esta clasificación fue creada para el trabajo y no corresponde a una clasificación oficial.

**Motores productivos:** industria, construcción, logística, forestal, pesca y agroalimentario.

**Capacidades transformadoras:** Digital/TIC, automatización/robótica y energía/sustentabilidad/biotecnología.


In [19]:
#Primero dejamos todas las carreras como "Otros campos"
d['ADN_BIOBIO'] = 'Otros campos'


**Explicación:** Antes de clasificar las carreras creamos una nueva columna llamada ADN_BIOBIO y asignamos inicialmente a todas el valor **“Otros campos”**. Esta es una forma práctica de comenzar la clasificación: después solo reemplazamos ese valor en las carreras que pertenecen a alguna de las familias estratégicas definidas para el trabajo. Esta celda no entrega una salida visible porque solamente prepara la columna que utilizaremos en las siguientes clasificaciones.


In [20]:
#Clasificamos las familias asociadas a motores productivos

#1. Industria y manufactura
industria = [
    "DISENO INDUSTRIAL",
    "ELECTRICIDAD INDUSTRIAL MENCION INSTALACIONES ELECTRICAS",
    "ELECTROMECANICA",
    "FABRICACION Y MONTAJE INDUSTRIAL",
    "INGENIERIA CIVIL DE MATERIALES",
    "INGENIERIA CIVIL EN METALURGIA",
    "INGENIERIA CIVIL INDUSTRIAL",
    "INGENIERIA CIVIL INDUSTRIAL - CONTINUIDAD INGENIEROS NO AFIN",
    "INGENIERIA CIVIL INDUSTRIAL, PROGRAMA CONTINUIDAD DE ESTUDIOS",
    "INGENIERIA CIVIL MECANICA",
    "INGENIERIA CIVIL METALURGICA",
    "INGENIERIA CIVIL QUIMICA",
    "INGENIERIA DE EJECUCION EN ELECTRICIDAD",
    "INGENIERIA DE EJECUCION EN ELECTRICIDAD-PE",
    "INGENIERIA DE EJECUCION EN GESTION INDUSTRIAL",
    "INGENIERIA DE EJECUCION EN MANTENIMIENTO INDUSTRIAL",
    "INGENIERIA DE EJECUCION EN MANTENIMIENTO INDUSTRIAL-PE",
    "INGENIERIA DE EJECUCION EN MECANICA",
    "INGENIERIA DE EJECUCION EN MECANICA AUTOMOTRIZ Y MAQUINARIA PESADA-PE",
    "INGENIERIA DE EJECUCION INDUSTRIAL",
    "INGENIERIA DE EJECUCION INDUSTRIAL PLAN CONTINUIDAD",
    "INGENIERIA DE EJECUCION INDUSTRIAL-PE",
    "INGENIERIA DE EJECUCION MECANICA",
    "INGENIERIA DE EJECUCION MECANICA DE PROCESOS Y MANTENIMIENTO INDUSTRIAL",
    "INGENIERIA DE EJECUCION MECANICA-PE",
    "INGENIERIA EN ELECTRICIDAD MENCION POTENCIA",
    "INGENIERIA EN ELECTRICIDAD MENCION PROYECTOS DE INSTALACIONES ELECTRICAS",
    "INGENIERIA EN MANTENIMIENTO INDUSTRIAL",
    "INGENIERIA EN MAQUINARIA PESADA Y VEHICULOS AUTOMOTRICES",
    "INGENIERIA EN MAQUINARIA, VEHICULOS AUTOMOTRICES Y SISTEMAS ELECTRONICOS",
    "INGENIERIA EN MECANICA AUTOMOTRIZ Y AUTOTRONICA",
    "INGENIERIA EN MECANICA Y ELECTROMOVILIDAD AUTOMOTRIZ",
    "INGENIERIA INDUSTRIAL",
    "INGENIERIA MECANICA EN MANTENIMIENTO INDUSTRIAL",
    "MANTENIMIENTO INDUSTRIAL",
    "MAQUINARIA PESADA",
    "MECANICA AUTOMOTRIZ EN MAQUINARIA PESADA",
    "MECANICA AUTOMOTRIZ EN SISTEMAS ELECTRONICOS",
    "MECANICA DE EQUIPO PESADO",
    "MECANICA INDUSTRIAL",
    "PROGRAMA DE CONTINUIDAD DE ESTUDIOS EN INGENIERIA DE EJECUCION INDUSTRIAL",
    "REFRIGERACION INDUSTRIAL Y AIRE ACONDICIONADO",
    "REFRIGERACION INDUSTRIAL Y CLIMATIZACION",
    "TECNICO DE NIVEL SUPERIOR EN MANTENCION INDUSTRIAL",
    "TECNICO DE NIVEL SUPERIOR EN MAQUINARIA NAVAL",
    "TECNICO EN ELECTRICIDAD",
    "TECNICO EN ELECTRICIDAD INDUSTRIAL",
    "TECNICO EN ELECTROMECANICA",
    "TECNICO EN FABRICACION Y MONTAJE INDUSTRIAL",
    "TECNICO EN MANTENIMIENTO INDUSTRIAL",
    "TECNICO EN MECANICA AUTOMOTRIZ Y AUTOTRONICA",
    "TECNICO EN MECANICA AUTOMOTRIZ Y MAQUINARIA PESADA",
    "TECNICO EN MECANICA AUTOMOTRIZ Y SISTEMAS ELECTRICOS",
    "TECNICO EN MECANICA Y ELECTROMOVILIDAD AUTOMOTRIZ",
    "TECNICO QUIMICO ANALISTA INDUSTRIAL",
    "TECNICO UNIVERSITARIO EN ELECTRICIDAD",
    "TECNICO UNIVERSITARIO EN MANTENIMIENTO INDUSTRIAL",
    "TECNICO UNIVERSITARIO EN MECANICA AUTOMOTRIZ",
    "TECNICO UNIVERSITARIO EN MECANICA INDUSTRIAL",
    "TECNICO UNIVERSITARIO EN QUIMICA CON MENCION EN QUIMICA INDUSTRIAL"
]

d.loc[
    d['NOMBRE CARRERA'].isin(industria),
    'ADN_BIOBIO'
] = 'Industria y manufactura'


#2. Construcción e infraestructura
construccion = [
    "ARQUITECTURA",
    "CONSTRUCCION",
    "CONSTRUCCION CIVIL",
    "CONSTRUCCION CIVIL (PLAN CONTINUIDAD)",
    "DIBUJO Y MODELAMIENTO ARQUITECTONICO Y ESTRUCTURAL",
    "INGENIERIA CIVIL",
    "INGENIERIA CIVIL-PLAN COMUN",
    "INGENIERIA EN CONSTRUCCION",
    "INGENIERIA EN CONSTRUCCION-PE",
    "PROGRAMA DE CONTINUIDAD DE ESTUDIOS EN CONSTRUCCION CIVIL",
    "TECNICO DE NIVEL SUPERIOR EN CONSTRUCCION",
    "TECNICO DE NIVEL SUPERIOR EN CONSTRUCCION CIVIL",
    "TECNICO EN CONSTRUCCION",
    "TECNICO EN CONSTRUCCIONES CIVILES",
    "TECNICO UNIVERSITARIO EN CONSTRUCCION"
]

d.loc[
    d['NOMBRE CARRERA'].isin(construccion),
    'ADN_BIOBIO'
] = 'Construccion e infraestructura'


#3. Logística y puertos
logistica = [
    "INGENIERIA CIVIL LOGISTICA",
    "INGENIERIA DE EJECUCION EN GESTION LOGISTICA-PE",
    "INGENIERIA DE EJECUCION EN LOGISTICA MARITIMA PORTUARIA",
    "INGENIERIA DE EJECUCION EN LOGISTICA MARITIMA PORTUARIA-PE",
    "INGENIERIA DE EJECUCION INDUSTRIAL MENCION MANTENIMIENTO Y LOGISTICA",
    "INGENIERIA EN COMERCIO EXTERIOR",
    "INGENIERIA EN GESTION LOGISTICA",
    "INGENIERIA EN LOGISTICA OPERATIVA",
    "INGENIERIA EN LOGISTICA Y TRANSPORTE",
    "LOGISTICA",
    "LOGISTICA Y OPERACIONES INDUSTRIALES",
    "TECNICO DE NIVEL SUPERIOR EN GESTION LOGISTICA",
    "TECNICO DE NIVEL SUPERIOR EN LOGISTICA Y OPERACIONES INDUSTRIALES",
    "TECNICO DE NIVEL SUPERIOR EN TRANSPORTE MARITIMO",
    "TECNICO EN ADMINISTRACION DE EMPRESAS MENCION LOGISTICA",
    "TECNICO EN COMERCIO EXTERIOR",
    "TECNICO EN GESTION LOGISTICA",
    "TECNICO EN LOGISTICA",
    "TECNICO EN LOGISTICA MARITIMA PORTUARIA",
    "TECNICO EN OPERACIONES LOGISTICAS",
    "TECNICO UNIVERSITARIO EN LOGISTICA"
]

d.loc[
    d['NOMBRE CARRERA'].isin(logistica),
    'ADN_BIOBIO'
] = 'Logistica y puertos'


#4. Forestal y madera
forestal = [
    "INGENIERIA CIVIL EN INDUSTRIAS DE LA MADERA",
    "INGENIERIA FORESTAL"
]

d.loc[
    d['NOMBRE CARRERA'].isin(forestal),
    'ADN_BIOBIO'
] = 'Forestal y madera'


#5. Pesca y acuicultura
pesca = [
    "ADMINISTRACION PORTUARIA PESQUERO ARTESANAL",
    "GESTION PORTUARIA Y PESQUERA",
    "INGENIERIA EN BIOTECNOLOGIA MARINA Y ACUICULTURA"
]

d.loc[
    d['NOMBRE CARRERA'].isin(pesca),
    'ADN_BIOBIO'
] = 'Pesca y acuicultura'


#6. Agroalimentario
agro = [
    "AGRONOMIA",
    "AGRONOMIA, PLAN CONTINUIDAD DE ESTUDIOS",
    "INGENIERIA AGRICOLA",
    "INGENIERIA DE EJECUCION AGROPECUARIA",
    "TECNICO AGRICOLA Y GANADERO",
    "TECNICO DE NIVEL SUPERIOR EN GESTION AGROINDUSTRIAL",
    "TECNICO EN VETERINARIA Y PRODUCCION PECUARIA",
    "TECNICO UNIVERSITARIO AGROPECUARIO",
    "TECNICO UNIVERSITARIO EN ALIMENTOS"
]

d.loc[
    d['NOMBRE CARRERA'].isin(agro),
    'ADN_BIOBIO'
] = 'Agroalimentario'


**Explicación:** En esta celda clasificamos las carreras relacionadas con los **motores productivos** de la región. Para cada familia definimos una lista de carreras y usamos isin() para buscar coincidencias en NOMBRE CARRERA. Cuando una carrera pertenece a una lista, reemplazamos “Otros campos” por la familia correspondiente. Aquí se incluyen industria y manufactura, construcción e infraestructura, logística y puertos, forestal y madera, pesca y acuicultura y agroalimentario. Esta clasificación fue creada para el trabajo y no corresponde a una clasificación oficial del Ministerio de Educación.


In [21]:
#Clasificamos las familias asociadas a capacidades transformadoras

#7. Digital y TIC
digital = [
    "ANALISTA PROGRAMADOR",
    "ANALISTA PROGRAMADOR COMPUTACIONAL",
    "COMPUTACION E INFORMATICA",
    "INFORMATICA BIOMEDICA",
    "INGENIERIA CIVIL EN INFORMATICA",
    "INGENIERIA CIVIL EN TELECOMUNICACIONES",
    "INGENIERIA CIVIL INFORMATICA",
    "INGENIERIA DE EJECUCION EN COMPUTACION E INFORMATICA",
    "INGENIERIA DE EJECUCION EN COMPUTACION E INFORMATICA-PE",
    "INGENIERIA DE EJECUCION EN INFORMATICA MENCION DESARROLLO DE SISTEMAS",
    "INGENIERIA EN CIBERSEGURIDAD",
    "INGENIERIA EN CONECTIVIDAD Y REDES",
    "INGENIERIA EN INFORMATICA",
    "INGENIERIA EN TELECOMUNICACIONES, CONECTIVIDAD Y REDES",
    "INGENIERIA GEOMATICA",
    "INGENIERIA INFORMATICA",
    "PROGRAMACION COMPUTACIONAL",
    "PROGRAMACION Y ANALISIS DE SISTEMAS",
    "TECNICO ANALISTA PROGRAMADOR",
    "TECNICO EN CONECTIVIDAD Y REDES",
    "TECNICO EN PROGRAMACION Y ANALISIS DE SISTEMAS",
    "TECNICO UNIVERSITARIO EN INFORMATICA",
    "TECNICO UNIVERSITARIO EN TELECOMUNICACIONES Y REDES",
    "TELECOMUNICACIONES, CONECTIVIDAD Y REDES"
]

d.loc[
    d['NOMBRE CARRERA'].isin(digital),
    'ADN_BIOBIO'
] = 'Digital y TIC'


#8. Automatización y robótica
automatizacion = [
    "AUTOMATIZACION Y CONTROL INDUSTRIAL",
    "ELECTRONICA",
    "ELECTRONICA INDUSTRIAL",
    "INGENIERIA CIVIL ELECTRONICA",
    "INGENIERIA CIVIL EN AUTOMATIZACION",
    "INGENIERIA DE EJECUCION ELECTRONICA",
    "INGENIERIA DE EJECUCION EN AUTOMATIZACION Y CONTROL-PE",
    "INGENIERIA DE EJECUCION EN CONTROL E INSTRUMENTACION INDUSTRIAL",
    "INGENIERIA DE EJECUCION EN ELECTRONICA",
    "INGENIERIA ELECTRONICA",
    "INGENIERIA EN AUTOMATIZACION Y CONTROL INDUSTRIAL",
    "INGENIERIA EN AUTOMATIZACION Y ROBOTICA",
    "INGENIERIA EN ELECTRICIDAD Y AUTOMATIZACION INDUSTRIAL",
    "INGENIERIA EN ELECTRICIDAD Y ELECTRONICA",
    "INGENIERIA EN ELECTRONICA Y SISTEMAS INTELIGENTES",
    "INSTRUMENTACION Y AUTOMATIZACION INDUSTRIAL",
    "INSTRUMENTACION Y CONTROL INDUSTRIAL",
    "TECNICO EN AUTOMATIZACION Y CONTROL INDUSTRIAL",
    "TECNICO EN AUTOMATIZACION Y ROBOTICA",
    "TECNICO EN ELECTRICIDAD Y AUTOMATIZACION INDUSTRIAL",
    "TECNICO EN ELECTRICIDAD Y ELECTRONICA",
    "TECNICO EN INSTRUMENTACION Y CONTROL",
    "TECNICO UNIVERSITARIO EN AUTOMATIZACION INDUSTRIAL",
    "TECNICO UNIVERSITARIO EN AUTOMATIZACION Y CONTROL",
    "TECNICO UNIVERSITARIO EN ELECTRONICA",
    "TECNICO UNIVERSITARIO EN ROBOTICA Y MECATRONICA"
]

d.loc[
    d['NOMBRE CARRERA'].isin(automatizacion),
    'ADN_BIOBIO'
] = 'Automatizacion y robotica'


#9. Energía, sustentabilidad y biotecnología
energia = [
    "ELECTRICIDAD Y EFICIENCIA ENERGETICA",
    "INGENIERIA AMBIENTAL",
    "INGENIERIA DE EJECUCION EN ENERGIAS RENOVABLES Y SUSTENTABILIDAD AMBIENTAL",
    "INGENIERIA DE EJECUCION EN GESTION Y CONTROL AMBIENTAL",
    "INGENIERIA DE EJECUCION EN QUIMICA Y MEDIO AMBIENTE-PE",
    "INGENIERIA EN BIOTECNOLOGIA VEGETAL",
    "INGENIERIA EN ELECTRICIDAD Y ENERGIAS RENOVABLES",
    "INGENIERIA EN ENERGIA Y SUSTENTABILIDAD AMBIENTAL",
    "INGENIERIA EN PREVENCION DE RIESGO Y MEDIOAMBIENTE",
    "QUIMICA AMBIENTAL",
    "TECNICO EN ENERGIAS RENOVABLES Y EFICIENCIA ENERGETICA",
    "TECNICO EN MEDIOAMBIENTE",
    "TECNICO UNIVERSITARIO EN CONTROL MEDIO AMBIENTE"
]

d.loc[
    d['NOMBRE CARRERA'].isin(energia),
    'ADN_BIOBIO'
] = 'Energia, sustentabilidad y biotecnologia'


**Explicación:** Esta celda aplica la misma lógica para las **capacidades transformadoras**. Se agrupan carreras vinculadas con **Digital y TIC**, **Automatización y robótica** y **Energía, sustentabilidad y biotecnología**. La idea es separar aquellas formaciones que pueden apoyar procesos de digitalización, automatización y modernización productiva. Igual que en la celda anterior, la clasificación es una decisión metodológica propia del proyecto y se utiliza únicamente para construir el análisis regional.


In [22]:
#Creamos los dos grandes bloques del análisis
motores = [
    'Industria y manufactura',
    'Construccion e infraestructura',
    'Logistica y puertos',
    'Forestal y madera',
    'Pesca y acuicultura',
    'Agroalimentario'
]

transformadoras = [
    'Digital y TIC',
    'Automatizacion y robotica',
    'Energia, sustentabilidad y biotecnologia'
]

d['BLOQUE_ADN'] = 'Otros campos'

d.loc[
    d['ADN_BIOBIO'].isin(motores),
    'BLOQUE_ADN'
] = 'Motores productivos'

d.loc[
    d['ADN_BIOBIO'].isin(transformadoras),
    'BLOQUE_ADN'
] = 'Capacidades transformadoras'


**Explicación:** Una vez asignadas las nueve familias, aquí las resumimos en dos bloques principales. Las seis familias productivas se guardan en la lista motores y las tres transformadoras en transformadoras. Luego creamos la columna BLOQUE_ADN, donde cada matrícula queda como **Motores productivos**, **Capacidades transformadoras** u **Otros campos**. Esta simplificación nos permite responder directamente la pregunta central del trabajo comparando la base productiva tradicional con las capacidades asociadas a la transformación tecnológica.


In [23]:
#Calculamos la frecuencia de los tres bloques
cuenta_bloque = d.groupby('BLOQUE_ADN').size()

tabla_bloque = pd.DataFrame({
    'Frecuencia absoluta': cuenta_bloque,
    'Frecuencia relativa (%)': cuenta_bloque/d.shape[0] * 100
})

round(tabla_bloque, 2)


,Frecuencia absoluta,Frecuencia relativa (%)
BLOQUE_ADN,,
Capacidades transformadoras,7000,6.92
Motores productivos,20368,20.15
Otros campos,73725,72.93


**Explicación:** Ahora contamos cuántas matrículas quedaron en cada uno de los tres bloques y calculamos sus porcentajes respecto de las 101.093 matrículas de pregrado. El resultado es **20.368 matrículas en Motores productivos (20,15%)**, **7.000 en Capacidades transformadoras (6,92%)** y **73.725 en Otros campos (72,93%)**. Estos porcentajes son el resultado principal de la clasificación regional y sirven como base para la comparación “Del acero al algoritmo”.


In [24]:
#Comparamos los dos bloques principales
tabla_acero_algoritmo = tabla_bloque.loc[
    ['Motores productivos', 'Capacidades transformadoras']
]

fig, ax = plt.subplots(figsize=(7,5))

ax.bar(
    tabla_acero_algoritmo.index,
    tabla_acero_algoritmo['Frecuencia relativa (%)'],
    edgecolor='black'
)

ax.set_title('Del acero al algoritmo')
ax.set_ylabel('Porcentaje de matrículas')

plt.show()


**Explicación:** Este gráfico compara solamente los dos bloques que nos interesan para la pregunta regional: motores productivos y capacidades transformadoras. Visualmente se observa que la barra de motores productivos es bastante más alta. Esto permite comunicar de manera sencilla la diferencia entre **20,15%** y **6,92%**. El gráfico describe la composición de la matrícula de 2021 y no debe interpretarse por sí solo como evidencia de déficit de profesionales o de atraso tecnológico.


In [25]:
#Calculamos cuántas veces es mayor el bloque productivo
razon = (
    tabla_acero_algoritmo.loc[
        'Motores productivos',
        'Frecuencia absoluta'
    ]
    /
    tabla_acero_algoritmo.loc[
        'Capacidades transformadoras',
        'Frecuencia absoluta'
    ]
)

round(razon, 1)


np.float64(2.9)

**Explicación:** Aquí calculamos una razón entre la cantidad de matrículas de ambos bloques. Dividimos las **20.368 matrículas de motores productivos** por las **7.000 transformadoras** y obtenemos aproximadamente **2,9**. Esto significa que, dentro de nuestra clasificación, en 2021 había cerca de **2,9 matrículas vinculadas a motores productivos por cada matrícula vinculada a capacidades transformadoras**. Esta razón resume en un solo número la diferencia mostrada en el gráfico anterior.


In [26]:
#Revisamos las familias estratégicas por separado
cuenta_adn = d[
    d['ADN_BIOBIO'] != 'Otros campos'
].groupby('ADN_BIOBIO').size()

tabla_adn = pd.DataFrame({
    'Frecuencia absoluta': cuenta_adn,
    'Frecuencia relativa (%)': cuenta_adn/d.shape[0] * 100
})

tabla_adn = tabla_adn.sort_values(
    'Frecuencia absoluta',
    ascending=False
)

round(tabla_adn, 2)


,Frecuencia absoluta,Frecuencia relativa (%)
ADN_BIOBIO,,
Industria y manufactura,11221,11.10
Construccion e infraestructura,6916,6.84
Digital y TIC,3365,3.33
Automatizacion y robotica,2608,2.58
Logistica y puertos,1046,1.03
"Energia, sustentabilidad y biotecnologia",1027,1.02
Agroalimentario,863,0.85
Forestal y madera,171,0.17
Pesca y acuicultura,151,0.15


**Explicación:** En esta celda dejamos de mirar solo los dos grandes bloques y revisamos las nueve familias por separado. **Industria y manufactura** es la familia con mayor peso, con **11.221 matrículas (11,10%)**, seguida de **Construcción e infraestructura (6,84%)**. Entre las capacidades transformadoras destacan **Digital y TIC (3,33%)** y **Automatización y robótica (2,58%)**. Esto permite entender qué familias explican la diferencia observada entre los dos bloques principales.


### Resultado principal

En 2021, **20,15%** de las matrículas de pregrado queda asociado a motores productivos y **6,92%** a capacidades transformadoras.

La relación es cercana a **2,9 a 1**. Esto describe la composición de la matrícula de 2021, pero no demuestra que exista déficit de profesionales.


# 5. Preguntas del trabajo


## Pregunta 1
### ¿Hay áreas del conocimiento donde las carreras sean más caras?

Usamos las **1.273 ofertas académicas únicas**.

**Criterio diseñado:** comparamos la **mediana del arancel de cada área** con la **mediana global** de las ofertas. Elegimos la mediana porque es menos sensible a valores extremos y representa mejor el arancel típico de un área.


In [27]:
#Agrupamos las ofertas por área del conocimiento
datos_p1 = ofertas.groupby(
    'AREA CONOCIMIENTO'
)['VALOR ARANCEL (PESOS)']

#Calculamos cantidad, media, mediana y cuartiles
tabla_p1 = pd.DataFrame({
    'Cantidad de ofertas': datos_p1.size(),
    'Media': datos_p1.mean(),
    'Mediana': datos_p1.median(),
    'Percentil 25': datos_p1.quantile(0.25),
    'Percentil 75': datos_p1.quantile(0.75)
})

tabla_p1 = tabla_p1.sort_values(
    'Mediana',
    ascending=False
)

mediana_global = ofertas[
    'VALOR ARANCEL (PESOS)'
].median()

print(f'Mediana global: ${mediana_global:.0f}')
round(tabla_p1, 0)


Mediana global: $2030000


,Cantidad de ofertas,Media,Mediana,Percentil 25,Percentil 75
AREA CONOCIMIENTO,,,,,
Derecho,18,3319065.0,3586000.0,2350000.0,4215000.0
Ciencias Basicas,31,3256166.0,3290000.0,2194500.0,4278045.0
Agropecuaria,18,3150941.0,2981500.0,1824250.0,4335000.0
Tecnologia,491,2301377.0,2122000.0,1730000.0,2276000.0
Salud,131,3044197.0,2080000.0,1781000.0,4203000.0
Humanidades,8,2115167.0,2006500.0,1717500.0,2579904.0
Arte y Arquitectura,35,2452345.0,1940000.0,1786000.0,2826500.0
Administracion y Comercio,251,2031019.0,1855000.0,1537000.0,2122000.0
Educacion,174,1952592.0,1722500.0,1399000.0,2654850.0


**Explicación:** Esta celda responde la primera pregunta del trabajo agrupando las **1.273 ofertas únicas** por área del conocimiento. Para cada área calculamos cantidad de ofertas, media, mediana, percentil 25 y percentil 75. Además obtenemos una **mediana global de $2.030.000** para usarla como referencia. Las medianas más altas aparecen en **Derecho ($3.586.000)**, **Ciencias Básicas ($3.290.000)** y **Agropecuaria ($2.981.500)**. Usamos principalmente la mediana porque representa el valor central y es menos sensible que la media frente a aranceles extremos.


In [28]:
#Graficamos la mediana del arancel por área
fig, ax = plt.subplots(figsize=(10,5))

ax.bar(
    tabla_p1.index,
    tabla_p1['Mediana']/1000000,
    edgecolor='black'
)

ax.set_title('Arancel mediano por área del conocimiento')
ax.set_ylabel('Millones de pesos')
plt.xticks(rotation=45, ha='right')

plt.show()


**Explicación:** El gráfico presenta la mediana del arancel de cada área en millones de pesos. Al ordenar previamente la tabla por mediana se observa con claridad que Derecho, Ciencias Básicas y Agropecuaria se ubican por encima de las demás áreas. La visualización complementa la tabla anterior y permite comparar rápidamente el arancel típico de cada área sin depender de todos los valores individuales.


In [29]:
#Cruzamos área del conocimiento y tipo de institución
tabla_p1_bivariada = pd.crosstab(
    ofertas['AREA CONOCIMIENTO'],
    ofertas['TIPO DE INSTITUCION'],
    ofertas['VALOR ARANCEL (PESOS)'],
    aggfunc='mean',
    margins=True,
    margins_name='Total'
)

round(tabla_p1_bivariada, 0)


TIPO DE INSTITUCION,Centros de Formacion Tecnica,Institutos Profesionales,Universidades (* Carrera en Convenio),Universidades CRUCH,Universidades Privadas,Total
AREA CONOCIMIENTO,,,,,,
Administracion y Comercio,1781338.0,1741044.0,NaN,2473294.0,2962050.0,2031019.0
Agropecuaria,1679333.0,1880000.0,NaN,4059110.0,4340417.0,3150941.0
Arte y Arquitectura,1769000.0,1850750.0,NaN,3848769.0,4637620.0,2452345.0
Ciencias Basicas,NaN,1878875.0,NaN,3758102.0,3495000.0,3256166.0
Ciencias Sociales,1353272.0,1419590.0,2327000.0,3691376.0,3069591.0,2129750.0
Derecho,NaN,1750000.0,2229000.0,4231569.0,3503926.0,3319065.0
Educacion,1364772.0,1439560.0,NaN,2673509.0,2490693.0,1952592.0
Humanidades,NaN,1005000.0,NaN,2763334.0,1929000.0,2115167.0
Salud,1735035.0,1740270.0,NaN,4419709.0,4246702.0,3044197.0


**Explicación:** Aquí construimos una tabla bivariada con crosstab() para cruzar **área del conocimiento** y **tipo de institución**. En cada combinación calculamos el arancel promedio. La tabla muestra que dentro de una misma área existen diferencias importantes dependiendo del tipo de institución. Por ejemplo, en varias áreas los promedios de universidades son mayores que los de CFT o IP. Por eso no interpretamos el área del conocimiento como la única característica asociada al precio.


### Respuesta

Sí. La mediana global de las ofertas es **$2.030.000**. Las medianas más altas corresponden a **Derecho ($3.586.000)**, **Ciencias Básicas ($3.290.000)** y **Agropecuaria ($2.981.500)**.

La tabla bivariada también muestra que el arancel cambia según el tipo de institución, por lo que el área no se interpreta como la única variable relacionada con el precio.


## Pregunta 2
### ¿Qué relación se observa entre la edad y el tipo de institución?

Primero resumimos la edad por tipo de institución y después comparamos los grupos de **15 a 19 años** y **40 años o más**.


In [30]:
#Calculamos cantidad, media y mediana de edad por tipo de institución
tabla_edad_institucion = d.groupby(
    'TIPO DE INSTITUCION'
)['EDAD'].agg(
    ['count', 'mean', 'median']
)

round(tabla_edad_institucion, 1)


,count,mean,median
TIPO DE INSTITUCION,,,
Centros de Formacion Tecnica,9455,25.7,23.0
Institutos Profesionales,26724,26.5,24.0
Universidades (* Carrera en Convenio),9,46.0,48.0
Universidades CRUCH,37997,22.8,22.0
Universidades Privadas,26908,24.0,23.0


**Explicación:** Para comenzar la segunda pregunta agrupamos las matrículas por tipo de institución y calculamos cantidad, edad media y edad mediana. Las **Universidades CRUCH** presentan una edad media de **22,8 años** y mediana de 22, mientras los **Institutos Profesionales** tienen una media de **26,5** y mediana de 24. Los CFT presentan una media de 25,7 años y las universidades privadas de 24 años. El grupo “Universidades (* Carrera en Convenio)” tiene solo 9 registros, por lo que no lo usamos como base para una conclusión general.


In [31]:
#Creamos una tabla de contingencia entre rango de edad e institución
tabla_edad_inst = pd.crosstab(
    d['RANGO EDAD'],
    d['TIPO DE INSTITUCION'],
    margins=True,
    margins_name='Total'
)

tabla_edad_inst


TIPO DE INSTITUCION,Centros de Formacion Tecnica,Institutos Profesionales,Universidades (* Carrera en Convenio),Universidades CRUCH,Universidades Privadas,Total
RANGO EDAD,,,,,,
15 a 19,1847,3570,0,8554,4527,18498
20 a 24,4075,11633,0,21963,14724,52395
25 a 29,1467,4842,0,5409,4826,16544
30 a 34,894,2826,2,1173,1477,6372
35 a 39,496,1829,1,482,652,3460
40 y mas,676,2024,6,416,702,3824
Total,9455,26724,9,37997,26908,101093


**Explicación:** Esta tabla de contingencia cruza los **rangos de edad** con el **tipo de institución**. Su objetivo es observar cómo se distribuyen los estudiantes de distintas edades entre CFT, IP, universidades CRUCH y universidades privadas. Por ejemplo, en el grupo de 15 a 19 años hay 8.554 matrículas CRUCH, mientras que en 40 años o más hay solo 416. La tabla completa permite ver que la composición institucional cambia bastante a medida que cambia el rango de edad.


In [32]:
#Analizamos el grupo de 15 a 19 años
grupo_15_19 = d[
    d['RANGO EDAD'] == '15 a 19 '
]

frecuencia_15_19 = grupo_15_19.groupby(
    'TIPO DE INSTITUCION'
).size()

tabla_15_19 = pd.DataFrame({
    'Frecuencia absoluta': frecuencia_15_19,
    'Frecuencia relativa (%)':
        frecuencia_15_19/grupo_15_19.shape[0] * 100
})

round(tabla_15_19, 1)


,Frecuencia absoluta,Frecuencia relativa (%)
TIPO DE INSTITUCION,,
Centros de Formacion Tecnica,1847,10.0
Institutos Profesionales,3570,19.3
Universidades CRUCH,8554,46.2
Universidades Privadas,4527,24.5


**Explicación:** Aquí aislamos el grupo de **15 a 19 años** y calculamos la distribución porcentual por tipo de institución. Las universidades CRUCH concentran **46,2%** del grupo, las universidades privadas **24,5%**, los Institutos Profesionales **19,3%** y los CFT **10,0%**. Por lo tanto, en el grupo más joven de la base las universidades, especialmente las CRUCH, tienen una presencia mayor.


In [33]:
#Analizamos el grupo de 40 años o más
grupo_40 = d[
    d['RANGO EDAD'] == '40 y mas'
]

frecuencia_40 = grupo_40.groupby(
    'TIPO DE INSTITUCION'
).size()

tabla_40 = pd.DataFrame({
    'Frecuencia absoluta': frecuencia_40,
    'Frecuencia relativa (%)':
        frecuencia_40/grupo_40.shape[0] * 100
})

round(tabla_40, 1)


,Frecuencia absoluta,Frecuencia relativa (%)
TIPO DE INSTITUCION,,
Centros de Formacion Tecnica,676,17.7
Institutos Profesionales,2024,52.9
Universidades (* Carrera en Convenio),6,0.2
Universidades CRUCH,416,10.9
Universidades Privadas,702,18.4


**Explicación:** Repetimos el mismo análisis para quienes tienen **40 años o más**. En este caso los **Institutos Profesionales concentran 52,9%**, seguidos por universidades privadas con 18,4% y CFT con 17,7%. Las universidades CRUCH representan solo 10,9%. Al compararlo con el grupo de 15 a 19 años aparece un cambio claro en la distribución institucional según el grupo de edad.


In [34]:
#Comparamos los dos grupos con gráficos de barras
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(13,5))

ax1.bar(
    tabla_15_19.index,
    tabla_15_19['Frecuencia relativa (%)'],
    edgecolor='black'
)

ax2.bar(
    tabla_40.index,
    tabla_40['Frecuencia relativa (%)'],
    edgecolor='black'
)

ax1.set_title('15 a 19 años')
ax2.set_title('40 años o más')
ax1.set_ylabel('Frecuencia relativa (%)')
ax2.set_ylabel('Frecuencia relativa (%)')

ax1.tick_params(axis='x', rotation=45)
ax2.tick_params(axis='x', rotation=45)

plt.tight_layout()
plt.show()


**Explicación:** Los dos gráficos colocan los grupos de **15 a 19 años** y **40 años o más** uno al lado del otro. Esta comparación visual permite observar inmediatamente el cambio de composición: en los más jóvenes predominan las universidades CRUCH, mientras que en el grupo de 40 años o más predominan los Institutos Profesionales. Esto demuestra una **asociación descriptiva** entre edad y tipo de institución, pero no permite afirmar que la edad sea la causa de la elección institucional.


### Respuesta

Entre **15 y 19 años**, las universidades CRUCH representan aproximadamente **46,2%** del grupo y los Institutos Profesionales **19,3%**.

Entre quienes tienen **40 años o más**, los Institutos Profesionales representan aproximadamente **52,9%** y las universidades CRUCH **10,9%**.

Se observa una asociación entre edad y tipo de institución, pero la base no permite afirmar que la edad sea la causa de la elección.


## Pregunta 3
### ¿Hay carreras cuyo arancel sea sustantivamente más caro que la mayoría?

Consideraremos como arancel sustantivamente alto a una oferta cuyo arancel esté **por sobre el percentil 90**.

Este criterio utiliza los percentiles trabajados en el Laboratorio 4: el percentil 90 deja aproximadamente al 90% de las ofertas en valores iguales o inferiores y permite estudiar el 10% superior.


In [35]:
#Calculamos el percentil 90 de las ofertas académicas
p90_ofertas = ofertas[
    'VALOR ARANCEL (PESOS)'
].quantile(0.90)

#También mostramos el valor máximo
maximo_ofertas = ofertas[
    'VALOR ARANCEL (PESOS)'
].max()

print(f'Percentil 90: ${p90_ofertas:.0f}')
print(f'Máximo: ${maximo_ofertas:.0f}')


Percentil 90: $4106400
Máximo: $8783670


**Explicación:** Para la tercera pregunta usamos el **percentil 90**, una medida de posición trabajada en el Laboratorio 4. El P90 de las ofertas académicas es aproximadamente **$4.106.400**. Esto significa que cerca del 90% de las ofertas tiene un arancel igual o inferior a ese valor y analizaremos las que quedan por encima.


In [36]:
#Filtramos las ofertas que superan el percentil 90
altas = ofertas[
    ofertas['VALOR ARANCEL (PESOS)'] > p90_ofertas
].copy()

cantidad_altas = altas.shape[0]
porcentaje_altas = cantidad_altas / ofertas.shape[0] * 100

print('Ofertas sobre P90:', cantidad_altas)
print(f'Porcentaje: {porcentaje_altas:.1f}%')


Ofertas sobre P90: 128
Porcentaje: 10.1%


**Explicación:** Al filtrar las ofertas que superan el percentil 90 obtenemos **128 ofertas de un total de 1.273**, equivalentes aproximadamente a **10,1%**. Es el grupo superior de aranceles que analizaremos a continuación.


In [37]:
#Revisamos dónde se concentran las ofertas de arancel alto
print('Por tipo de institución')
print(
    altas.groupby('TIPO DE INSTITUCION')
    .size()
    .sort_values(ascending=False)
)

print('\nPor área del conocimiento')
print(
    altas.groupby('AREA CONOCIMIENTO')
    .size()
    .sort_values(ascending=False)
)

print('\nPor provincia')
print(
    altas.groupby('PROVINCIA SEDE')
    .size()
    .sort_values(ascending=False)
)


Por tipo de institución
TIPO DE INSTITUCION
Universidades CRUCH       73
Universidades Privadas    55
dtype: int64

Por área del conocimiento
AREA CONOCIMIENTO
Salud                        38
Tecnologia                   38
Ciencias Basicas             13
Administracion y Comercio    10
Ciencias Sociales            10
Agropecuaria                  7
Derecho                       7
Arte y Arquitectura           5
dtype: int64

Por provincia
PROVINCIA SEDE
Concepcion    123
Biobio          5
dtype: int64


**Explicación:** Entre las 128 ofertas sobre P90 encontramos **73 ofertas CRUCH y 55 privadas**. **Salud y Tecnología concentran 38 ofertas cada una**. Territorialmente, **123** se encuentran en la provincia de Concepción y **5** en Biobío. Estos resultados son descriptivos y no prueban causalidad.


In [38]:
#Comparamos la duración mediana de las ofertas altas con el resto
duracion_altas = altas[
    'DURACION TOTAL CARRERA (SEMESTRES)'
].median()

resto = ofertas[
    ofertas['VALOR ARANCEL (PESOS)'] <= p90_ofertas
]

duracion_resto = resto[
    'DURACION TOTAL CARRERA (SEMESTRES)'
].median()

print(f'Duración mediana ofertas altas: {duracion_altas} semestres')
print(f'Duración mediana resto: {duracion_resto} semestres')


Duración mediana ofertas altas: 10.0 semestres
Duración mediana resto: 5.0 semestres


**Explicación:** Esta celda compara la duración de las ofertas con arancel alto frente al resto. La **duración mediana del grupo alto es de 10 semestres**, mientras que en las demás ofertas la mediana es de **5 semestres**. La diferencia es importante y ayuda a caracterizar el grupo de precios altos. Sin embargo, la comparación es descriptiva: no significa que una carrera sea más cara solamente por durar más semestres.


In [39]:
#Calculamos el arancel mediano por carrera
ranking_carreras = ofertas.groupby(
    'NOMBRE CARRERA'
)['VALOR ARANCEL (PESOS)'].agg(
    ['count', 'median']
)

ranking_carreras.columns = [
    'Cantidad de ofertas',
    'Arancel mediano'
]

#Consideramos carreras con al menos 3 ofertas
ranking_carreras = ranking_carreras[
    ranking_carreras['Cantidad de ofertas'] >= 3
]

ranking_carreras = ranking_carreras.sort_values(
    'Arancel mediano',
    ascending=False
).head(10)

ranking_carreras


,Cantidad de ofertas,Arancel mediano
NOMBRE CARRERA,,
MEDICINA,3,7490000.0
ODONTOLOGIA,5,7290000.0
INGENIERIA CIVIL QUIMICA,3,4804479.0
TECNOLOGIA MEDICA,4,4771050.0
INGENIERIA CIVIL INFORMATICA,3,4745935.0
MEDICINA VETERINARIA,6,4320000.0
KINESIOLOGIA,9,4224000.0
ARQUITECTURA,6,4214450.0
INGENIERIA CIVIL ELECTRICA,4,4202239.5


**Explicación:** Finalmente agrupamos las ofertas por nombre de carrera y calculamos cuántas ofertas tiene cada carrera y cuál es su arancel mediano. Para evitar rankings basados en una sola observación, dejamos solamente carreras con **al menos tres ofertas**. El resultado muestra a **Medicina** con una mediana de aproximadamente **$7.490.000** y **Odontología** con **$7.290.000**, seguidas por varias ingenierías y Tecnología Médica. Esta tabla complementa la pregunta 3 mostrando ejemplos concretos de carreras que se ubican entre las de mayor arancel mediano.


### Respuesta

Sí. El **percentil 90** de las 1.273 ofertas es aproximadamente **$4.106.400**.

Superan ese valor **128 ofertas**, equivalentes a aproximadamente **10,1%** del total.

De esas ofertas, **73** pertenecen a universidades CRUCH y **55** a universidades privadas. **Salud** y **Tecnología** concentran 38 cada una y la mayoría se ubica en la provincia de **Concepción**.

La duración mediana del grupo de arancel alto es de **10 semestres**, frente a **5 semestres** en el resto. Entre las carreras con mayor arancel mediano aparecen **Medicina** y **Odontología**.

**Explicación encontrada:** dentro de la base, las ofertas sobre P90 se concentran en universidades, especialmente en las áreas de **Salud** y **Tecnología**, presentan una duración mediana mayor y se ubican casi por completo en la provincia de **Concepción**. Estas características ayudan a describir el perfil del grupo de mayor arancel y la concentración en Concepción aporta el contexto territorial de la región. Sin embargo, son asociaciones descriptivas y **no demuestran que estas variables sean la causa directa del precio**.


# 6. Conclusiones

- La base original contiene **106.555 registros** y el análisis principal utiliza **101.093 matrículas de pregrado**.
- Para comparar precios trabajamos con **1.273 ofertas académicas únicas**.
- Los motores productivos representan **20,15%** de las matrículas de pregrado y las capacidades transformadoras **6,92%**.
- La relación entre ambos bloques es aproximadamente **2,9 a 1**.
- Las tres preguntas del trabajo se responden usando tablas de frecuencia, medidas de tendencia central, gráficos, tablas bivariadas y medidas de dispersión.

# 7. Limitaciones

- La base corresponde solo a 2021.
- La clasificación “Del acero al algoritmo” fue creada para este trabajo.
- Los resultados son descriptivos.
- La base no permite demostrar causalidad.
- No contiene salarios, vacantes, productividad ni demanda laboral.
